onvif/src/discovery/mod.rs

```diff
@@ -186,20 +215,51 @@ where
                 .is_some()
         })
         .flat_map(|probe_match| {
-            probe_match
-                .x_addrs()
-                .into_iter()
-                .zip(iter::repeat(probe_match.name()))
+            let mut urls = probe_match.x_addrs();
+            let name = probe_match.name();
+
+            urls.retain(|url| match url.host() {
+                // Filter out link-local IP addresses.
+                // TODO: Make filtering link-local addresses configurable.
+                Some(url::Host::Ipv4(ip)) if ip.is_link_local() => {
+                    debug!("{name:?}: Filtering out link-local IP address: {url}");
+                    false
+                }
+                // Filter out IPv6 addresses. Camera's usually have both, so let's use only IPv4.
```

Comment:
You should account for cases where there is only an IPv6. I've seen this happen on some networks.